internal/npm_install/node_module_library.bzl

```diff
@@ -0,0 +1,70 @@
+# Copyright 2019 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+load("@build_bazel_rules_nodejs//internal/common:node_module_info.bzl", "NodeModuleInfo", "NodeModuleSources")
+load("@build_bazel_rules_nodejs//internal/common:providers.bzl", "ScriptsProvider")
+
+def _node_module_library_impl(ctx):
+    apf_umds = []
+    apf_factories = []
+    apf_summaries = []
+
+    # If this npm package is in the Angular package format then collect
+    # umd bundles, ngfactory files & ngsummary files and provide them
+    # via the ScriptsProvider
+    if ctx.attr.is_apf:
```

Comment:
do we really need to distinguish apf libraries at this point? it seems like we want the NodeModuleSources to give us a list of scripts to include in the ScriptsProvider, we shouldn't need to enumerate them here. Colud the `.(umd|ngfactory|ngsummary).js$` logic go in the place where the `node_module_library` was declared, and be passed with a `devscripts` attribute or something?

Reply:
Yes. That sounds much cleaner. 👍 👍

Reply:
For that matter, do we want a separate `ScriptsProvider` or just provide `scripts` in the `NodeModulesSources` provider?

Reply:
done